cpp/tests/io/json/json_utils.hpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "io/json/read_json.hpp"
+
+#include <cudf/io/datasource.hpp>
+#include <cudf/io/json.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <numeric>
+
+// Helper function to test correctness of JSON byte range reading.
+// We split the input source files into a set of byte range chunks each of size
+// `chunk_size` and return an array of partial tables constructed from each chunk
+template <typename idx = std::int32_t>
+std::vector<cudf::io::table_with_metadata> split_byte_range_reading(
```

Comment:
This function always reads all inputs? If that's the case, we could read all data into a host buffer and pass `record_ranges` spans of this buffer in each `read_json` call. This way we avoid the use for byte_range in this code, making it less likely to get matching incorrect behavior with the tested code.
If this blows up the current implementation, feel free to ignore.

Reply:
I'm not sure I follow - wouldn't we miss checking the correctness of byte range if we directly provide the host buffer range (which may span multiple sources)?

Reply:
resolved based on offline discussion - I misunderstood the purpose of this function.